### PR TITLE
Changes to build on Alpine Linux.

### PR DESCRIPTION
**Pull request by daira**
ref #1558

Signed-off-by: Daira Hopwood <daira@jacaranda.org>

### CODE REVIEW

**Comment by daira** [comment]
Note that [<poll.h>](http://pubs.opengroup.org/onlinepubs/7990989799/xsh/poll.h.html) and [<fcntl.h>](http://pubs.opengroup.org/onlinepubs/7908799/xsh/fcntl.h.html) are the standard POSIX locations for those headers.

**Comment by str4d** [comment]
ACK

**Comment by daira** [comment]
This doesn't seem to be solving #1558 for the original reporter, and I need to dig into the build output to figure out why. Do not merge yet.

**Comment by str4d** [comment]
Apologies for not spotting this in my 10pm review:you need to manually add the patch commands. See the existing patch command in `define $(package)_preprocess_cmds`.

**Comment by daira** [comment]
Ah ok. I've just gone to bed, can someone else add those please?

**Comment by daira** [comment]
This should be fixed now.

File: depends/packages/boost.mk [hunk 1]
```diff
@@ -27,7 +27,8 @@ endef
 
 define $(package)_preprocess_cmds
   echo "using $(boost_toolset_$(host_os)) : : $($(package)_cxx) : <cxxflags>\"$($(package)_cxxflags) $($(package)_cppflags)\" <linkflags>\"$($(package)_ldflags)\" <archiver>\"$(boost_archiver_$(host_os))\" <striper>\"$(host_STRIP)\"  <ranlib>\"$(host_RANLIB)\" <rc>\"$(host_WINDRES)\" : ;" > user-config.jam && \
-   patch -p1 < $($(package)_patch_dir)/deprecated_auto_ptr.patch
+  patch -p1 < $($(package)_patch_dir)/deprecated_auto_ptr.patch
```
Comment (str4d) [inline depends/packages/boost.mk]: Missing `&& \` at end.

**Comment by daira** [comment]
Comments addressed, please re-review.

**Comment by daira** [comment]
@zkbot r+

**Pull request merged**